src/integrationTest/java/com/hivemq/helmcharts/util/OperatorHelmChartContainer.java

```diff
@@ -1,85 +1,164 @@
 
 /**
  * Container that includes the helm binary to be able to install the HiveMQ helm charts
  */
 public class OperatorHelmChartContainer extends K3sContainer {
     public final static int mqttPort = 1883;
+    private final @NotNull List<String> imagesNames;
+    private boolean withCustomImages = false;
 
-    private OperatorHelmChartContainer(final @NotNull String k3sVersion,
-                                       final @NotNull File dockerfile,
-                                       final @NotNull File helmChartMountPath,
-                                       final @NotNull String containerPath) {
-        super(getAdHocImageName(k3sVersion, dockerfile));
-        final MountableFile helmChartMountable = MountableFile.forHostPath(helmChartMountPath.getPath());
-        super.withCopyFileToContainer(helmChartMountable, containerPath);
+    public OperatorHelmChartContainer(final @NotNull String k3sVersion,
+                                      final @NotNull String dockerfileName,
+                                      final @NotNull String customValuesFile) {
+        super(getAdHocImageName(k3sVersion, dockerfileName));
         super.addExposedPort(mqttPort);
+        super.withFileSystemBind("./charts/hivemq-operator", "/chart");
+        super.withCopyFileToContainer(MountableFile.forClasspathResource(customValuesFile), "/files/values.yml");
+        super.withStartupCheckStrategy(new DeploymentStatusStartupCheckStrategy(this));
+        imagesNames = new ArrayList<>();
     }
 
-    public static @NotNull DockerImageName getAdHocImageName(final @NotNull String k3sVersion,
-                                                             final @NotNull File dockerfile) {
-        final String s = new ImageFromDockerfile().withDockerfile(
-                        dockerfile.toPath())
+    private static @NotNull DockerImageName getAdHocImageName(final @NotNull String k3sVersion,
+                                                             final @NotNull String dockerfileName) {
+        var dockerfile = new File(MountableFile.forClasspathResource(dockerfileName).getFilesystemPath());
+
+        final String s = new ImageFromDockerfile().withDockerfile(dockerfile.toPath())
                 .withBuildArg("K3S_VERSION", k3sVersion)
                 .get();
-        return DockerImageName.parse(s).asCompatibleSubstituteFor("rancher/k3s");
-    }
 
-    public static @NotNull Builder builder() {
-        return new Builder();
+        return DockerImageName.parse(s).asCompatibleSubstituteFor("rancher/k3s");
     }
 
-    public int getMappedPort() {
-        return super.getMappedPort(mqttPort);
+    /**
+     * Uses custom images instead of docker hub images, additional images can be added
+     */
+    public @NotNull OperatorHelmChartContainer withCustomImages(@Nullable String... fileNames) {
+        withCustomImages = true;
+        super.withFileSystemBind("./build/containers", "/build", BindMode.READ_ONLY);
```

Comment:
Use `withCopyFileToContainer` instead of `withFileSystemBind`. `withCopyFileToContainer` seems to be more stable on CIs.